legate/core/operation.py

```diff
@@ -195,14 +209,29 @@ def add_dtype_arg(self, dtype):
     def add_future(self, future):
         self._futures.append(future)
 
+    def add_fusion_metadata(self, fusion_metadata):
+        self._is_fused = True
+        self._fusion_metadata = fusion_metadata
+
     def launch(self, strategy):
         launcher = TaskLauncher(self.context, self._task_id, self.mapper_id)
 
-        for input, input_part in zip(self._inputs, self._input_parts):
+        if self._is_fused:
+            launcher.add_fusion_metadata(self._is_fused, self._fusion_metadata)
+        if  self._is_fused: #fused ops re-use encapsulated unfused partitions
+            input_parts = self._unfused_input_parts
+            output_parts = self._unfused_output_parts
+            reduction_parts = self._unfused_reduction_parts
+        else:
+            input_parts = self._input_parts
+            output_parts = self._output_parts
+            reduction_parts = self._reduction_parts
```

Comment:
Currently, a hash method is used to map a store (or rather it's constraints) to it's partition. all hasehes partition constraints (for inputs, outputs, reductions) are ingest their task's absolute id, which is problematic for fusion as the fused task will have a different absolute id than the unfused task. 
To take care of  this we set `input_parts` to `self._unfused_input_parts`, so we can hash into the original unfused partitions. Otherwise, the method for hashing into a store's partition will complain.